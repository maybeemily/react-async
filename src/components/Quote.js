import React from 'react';
import PropTypes from 'prop-types';

function Quote({ quote }){
  return (
    <section>
      <img src={quote.image}/>
      <p>{quote.character}</p>
      <p>{quote.quote}</p>
    </section>
  );
}

Quote.propTypes = {
  quote: PropTypes.shape({
    character: PropTypes.string.isRequired,
    quote: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired
  }).isRequired
};

export default Quote;
