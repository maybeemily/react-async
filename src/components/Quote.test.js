import React from 'react';
import { shallow } from 'enzyme';
import Quote from './Quote';

describe('Quote component', () => {
  it('renders the quote', () => {
    const testQuote = {
      character: 'Leela',
      quote: 'Awww',
      image: '/path'
    };
    const wrapper = shallow(<Quote quote={testQuote} />);
    expect(wrapper).toMatchSnapshot();
  });
});
