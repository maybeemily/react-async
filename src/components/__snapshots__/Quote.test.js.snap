// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote component renders the quote 1`] = `
<section>
  <img
    src="/path"
  />
  <p>
    Leela
  </p>
  <p>
    Awww
  </p>
</section>
`;
